hqmedia.upload_controller.js

```diff
@@ -289,6 +289,9 @@ function BaseHQMediaUploadController (uploader_name, marker, options) {
         if (!self.isMultiFileUpload) {
             var newExtension = '.' + self.filesInQueueUI[0].get('name').split('.').pop().toLowerCase();
             self.uploadParams.path = self.uploadParams.path.replace(/\.[^/.]+$/, newExtension);
```

Comment:
This change could be incorporated into this line by changing the regex to `/(\.[^/.]+)?$/` (and delete the following three lines)